ledger/ledger-on-sql/src/test/suite/scala/com/daml/ledger/on/sql/RestartSpec.scala

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2020 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.on.sql
+
+import java.nio.file.Files
+import java.util.UUID
+
+import akka.stream.scaladsl.Sink
+import com.daml.ledger.participant.state.kvutils.api.KeyValueParticipantState
+import com.daml.ledger.participant.state.v1.ParticipantId
+import com.daml.ledger.participant.state.v1.Update.PartyAddedToParticipant
+import com.digitalasset.daml.lf.data.Ref
+import com.digitalasset.daml.lf.data.Ref.{LedgerString, Party}
+import com.digitalasset.ledger.api.testing.utils.AkkaBeforeAndAfterAll
+import com.digitalasset.logging.LoggingContext.newLoggingContext
+import com.digitalasset.resources.Resource
+import org.scalatest.{AsyncWordSpec, BeforeAndAfterEach, Matchers}
+
+import scala.compat.java8.FutureConverters._
+import scala.concurrent.duration.DurationInt
+import scala.util.Random
+
+class RestartSpec
+    extends AsyncWordSpec
+    with Matchers
+    with BeforeAndAfterEach
+    with AkkaBeforeAndAfterAll {
+
+  private val root = Files.createTempDirectory(getClass.getSimpleName)
+
+  private def start(id: String): Resource[KeyValueParticipantState] = {
+    newLoggingContext { implicit logCtx =>
```

Comment:
if you want you can also use `newLoggingContext` after the `in` method when defining tests. Whatever you feel makes more sense.

Reply:
I'll leave it here for now but I'll consider it when this gets refactored into `ParticipantStateIntegrationSpecBase`.